aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,767 @@
+            ]
+
+        """
+        shape_list = [{'T': (0, img.nt),
+                       'C': (0, img.channels),
+                       'Z': (0, img.nz),
+                       'Y': (0, img.dims[1]),
+                       'X': (0, img.dims[0])}
+                      for idx, img in enumerate(lif.get_iter_image())]
+        consistent = all(elem == shape_list[0] for elem in shape_list)
+        if consistent:
+            shape_list[0]['S'] = (0, len(shape_list))
+            shape_list = [shape_list[0]]
+        else:
+            for idx, lst in enumerate(shape_list):
+                lst['S'] = (idx, idx+1)
+        return shape_list
+
+    @staticmethod
+    def _read_dims_to_ranges(lif: LifFile, read_dims: Optional[Dict[str, int]] = None):
+        """
+        Convert the provided read_dims and file structure into ranges to iterate over
+
+        Parameters
+        ----------
+        lif: LifFile
+            The LifFile to get the ranges from
+        read_dims: Dict[str: int]
+            The list of locked dimensions
+
+        Returns
+        -------
+        Dict[Dimension: range]
+            These ranges can then be used to iterate through the specified YX images
+
+        """
+        if read_dims is None:
+            read_dims = {}
+
+        data_shape = LifReader._dims_shape(lif=lif)
+
+        # if S is in read_dims then use the specified value and the specified dims for that scene
+        if Dimensions.Scene in read_dims:
+            s_range = range(read_dims[Dimensions.Scene], read_dims[Dimensions.Scene] + 1)
+            s_dict = data_shape[s_range[0]]
+        else:
+            s_range = range(*data_shape[0][Dimensions.Scene])
+            s_dict = data_shape[0]
+
+        # map the dims over to ranges and if the dim is in read_dims make the range over the single dim
+        integrated_dims = {Dimensions.Scene: s_range}
+        for dim in [Dimensions.Time, Dimensions.Channel, Dimensions.SpatialZ]:
+            if dim in read_dims:
+                integrated_dims[dim] = range(read_dims[dim], read_dims[dim]+1)
+            else:
+                integrated_dims[dim] = range(*s_dict[dim])
+
+        return integrated_dims
+
+    @staticmethod
+    def get_pixel_type(meta: Element, scene: int = 0) -> np.dtype:
+        """
+        This function parses the metadata to assign the appropriate numpy.dtype
+
+        Parameters
+        ----------
+        meta: lxml.etree.Element
+            The root Element of the metadata etree
+        scene: int
+            The index of the scene, scenes could have different storage data types.
+
+        Returns
+        -------
+        numpy.dtype
+            The appropriate data type to construct the matrix with.
+
+        """
+        ############################
+        #
+        #  Due to the 12 bit values being stored in a uint16 the raw data is a little fussy to get the
+        #  contrast correct.
+        #
+        ############################
```

Comment:
Whyyy?? Just make this a two line comment?

Reply:
The formatting was a hangover from when the comment said something else that I wanted to draw attention to because it was weird.